### PR TITLE
GEOMESA-1902 Fixing secondary upper ranges in attribute index

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

**Comment by jnh5y** [comment]
Uhmm...  by chance, is there a unit test that could cover this change?  I only ask since this is a tad subtle (at least for my vacation recovering brain).

**Comment by elahrvivaz** [comment]
that would be a good idea :)

File: geomesa-index-api/src/test/scala/org/locationtech/geomesa/index/TestGeoMesaDataStore.scala [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+                           ecql: Option[Filter]) extends TestQueryPlanType {
+    override def scan(ds: TestGeoMesaDataStore): CloseableIterator[SimpleFeature] = {
+      def contained(range: TestRange, row: Array[Byte]): Boolean =
+        byteComparator.compare(range.start, row) <= 0 && byteComparator.compare(range.end, row) > 0
+      index.features.toIterator.collect { case (row, sf) if ranges.exists(contained(_, row))  => sf }
+    }
+
+    override def explain(explainer: Explainer, prefix: String): Unit = {
+      explainer(s"ranges (${ranges.length}): ${ranges.take(5).map(r => s"[${r.start.mkString("")}:${r.end.mkString("")})")}")
+      explainer(s"ecql: ${ecql.map(org.locationtech.geomesa.filter.filterToString).getOrElse("INCLUDE")}")
```
Comment (jnh5y) [inline geomesa-index-api/src/test/scala/org/locationtech/geomesa/index/TestGeoMesaDataStore.scala]: I started to feel like a horrible person for asking for this test given that you implemented an entire in-memory GeoMesa DataStore.  

The fact that you implemented an 'explain' method stopped that feeling completely.;)  Well done!

Comment (elahrvivaz) [inline geomesa-index-api/src/test/scala/org/locationtech/geomesa/index/TestGeoMesaDataStore.scala]: I'd been thinking about doing it for a while, so you gave me the excuse :)

**Pull request merged**